packages/core/src/baseTransfer.ts

```diff
@@ -0,0 +1,86 @@
+import type { Config } from './config/config.js';
+import type { Domainlike, EvmResource, Domain, SubstrateResource } from './types.js';
+
+export interface BaseTransferParams {
+  source: Domainlike;
+  destination: Domainlike;
+  resource: string | EvmResource | SubstrateResource;
+  sourceAddress: string;
+}
+
+export abstract class BaseTransfer {
+  protected _destination: Domain;
+  protected _resource: EvmResource | SubstrateResource;
+  protected _source: Domain;
+  protected _config: Config;
+
+  protected sourceAddress: string;
+
+  public get source(): Domain {
+    return this.source;
+  }
+
+  public get destination(): Domain {
+    return this.destination;
+  }
+
+  public get resource(): EvmResource | SubstrateResource {
+    return this.resource;
+  }
+
+  public get config(): Config {
+    return this._config;
+  }
+
+  private findResource(
+    resource: string | EvmResource | SubstrateResource,
+  ): EvmResource | SubstrateResource | undefined {
+    return this._config.getResources(this.source).find(_resource => {
+      return typeof resource === 'string'
+        ? resource === _resource.resourceId
+        : resource.resourceId === _resource.resourceId;
+    });
+  }
+
+  constructor(transfer: BaseTransferParams, config: Config) {
+    this.sourceAddress = transfer.sourceAddress;
+    this._source = config.getDomain(transfer.source);
+    this._destination = config.getDomain(transfer.destination);
+    const __resource = this.findResource(transfer.resource);
```

Comment:
I think here you can just rename `__resource` to something else to avoid the `__` double

Reply:
already discussed

Reply:
@wainola what is the result of the discussion? I would avoid such variable names, as an alternative if it's a value inside the scope of constructor is better to  name `const resourceValue = ` or any other alternative